mainhandler/handlerequests.go

```diff
@@ -77,6 +78,47 @@ func NewActionHandler(k8sAPI *k8sinterface.KubernetesApi, sessionObj *utils.Sess
 	}
 }
 
+func (mainHandler *MainHandler) HandleWatchers(ctx context.Context, scanOnNewImg bool) {
+	defer func() {
+		if err := recover(); err != nil {
+			logger.L().Ctx(ctx).Error(fmt.Sprintf("RECOVER in HandleWatchers, reason: %v", err))
+		}
+	}()
+
+	watchHandler := watcher.NewWatchHandler()
+
+	// build imageIDs and wlids maps
+	err := watchHandler.Initialize(scanOnNewImg)
+	if err != nil {
+		logger.L().Ctx(ctx).Error(err.Error(), helpers.Error(err))
+		return
+	}
+
+	// scan all workloads in cluster
+	mainHandler.TriggerScanForWorkloads(ctx, watchHandler.GetWlidsMap())
+
+	// start watching
+	watchHandler.PodWatch(ctx, scanOnNewImg)
+}
+
+func (mainHandler *MainHandler) TriggerScanForWorkloads(ctx context.Context, wlidsToContainerToImageID map[string]map[string]string) {
+	go func() {
```

Comment:
Let's move the logic outside go routine since we are iterating the `wlidsToContainerToImageID` (map is a pointer type) while changing it in the `PodWatch` (line 101).

In order to avoid sending the same command twice, we can build all list of `Command{}` in this function, and only move this logic to a go routine:
```
newSessionObj := utils.NewSessionObj(ctx, cmd, "Websocket", "", uuid.NewString(), 1)
*mainHandler.sessionObj <- *newSessionObj
```